network/mq/mq.go

```diff
@@ -82,7 +135,33 @@ func (q Queue) GetMessageChannel() MessageChannel {
 	return messageChannel
 }
 
-func (q Queue) RunConsumerWithCancel(consumer Consumer, async bool, ctx context.Context) {
+func worker(messages <-chan amqp.Delivery, consumer Consumer) {
+	for msg := range messages {
+		err := consumer.Callback(msg)
+		if err != nil {
+			log.Error(err)
+		}
+		//if options.RetryOnError {
```

Comment:
@vikmeup remove this?

Reply:
let me keep it for now, I will work on this little later